### ISSUE TITLE
Potential Data Leakage in Preprocessing and Analysis of Predictive Power

### ISSUE THREAD

**Issue by zahravalidi** [124015001/healthcare-insurance-fraud_detection#4]
Hello Abhimanyu,

First, thank you for providing this valuable dataset and the accompanying notebook. It's a great resource for working with real-world medical claims data.

I have been conducting a detailed analysis of the dataset and the methodology in the Singh,_Abhimanyu_(2023)_code.ipynb notebook. During this process, I came across two key findings that I wanted to share, as they might help improve the example for future users.

1. Data Leakage in the Preprocessing Pipeline
In the original notebook, the SMOTE (Synthetic Minority Over-sampling Technique) is applied to the entire dataset before the data is split into training and testing sets.

Issue: This workflow can lead to data leakage. When SMOTE creates synthetic samples, it uses information from the entire dataset. Consequently, the test set is no longer completely "unseen" by the training process, as its characteristics have influenced the creation of the synthetic training samples. This can lead to an inflated and unreliable accuracy score.

Suggestion: The standard practice to prevent this is to perform the train-test split first, and then apply SMOTE only to the training data (X_train, y_train).

2. Analysis of the Dataset's Predictive Power
After correcting for the data leakage by following the standard workflow (Split -> Preprocess -> Train), my analysis suggests that the features in this dataset have very low predictive power for the fraud target variable. Here is the evidence:

A. Univariate Analysis Shows No Clear Distinction

The distributions of key features are nearly identical for both fraudulent and non-fraudulent claims.

Numerical Features: As seen in the box plots, the distributions for Fee Charged and membership_period do not show a significant difference between the two classes.
<img width="1790" height="480" alt="Image" src="https://github.com/user-attachments/assets/96947f0e-ff2e-41c0-bbbf-123bcd74750c" />

Categorical Features: Similarly, the fraud rates across different categories like gender, location, and cause are all very close to the overall average of ~20%, indicating they do not strongly differentiate between the two outcomes.
<img width="1990" height="480" alt="Image" src="https://github.com/user-attachments/assets/62d3cbad-8068-4622-8690-0ac445252bde" />

B. Model Performance is Equivalent to Random Chance

I trained a standard XGBoost model on the preprocessed data (using the corrected, leak-free workflow). The final performance on the held-out test set was:

AUC-ROC Score: 0.4990

An AUC score of ~0.5 indicates that the model has no more predictive power than a random guess.

C. Permutation Importance Confirms Lack of Signal

To verify this, I ran a permutation importance analysis. This technique measures how much a model's performance drops when a single feature's values are shuffled. The results show that shuffling the features leads to a drop in the AUC score that is centered around zero, which strongly confirms that the features do not provide a significant predictive signal.
<img width="738" height="455" alt="Image" src="https://github.com/user-attachments/assets/34c6c58b-8fcb-454c-93a1-604c561e39a1" />

Conclusion & Additional Data
The original notebook is a fantastic exercise in data cleaning and feature engineering. However, the data leakage from the SMOTE implementation leads to an optimistic performance score. Once this is corrected, the underlying data appears to lack the necessary signal to predict the fraud label with this feature set. I would recommend updating the notebook to reflect the correct train-then-resample workflow.

This problem of fraud detection in healthcare is incredibly important, and your dataset has been an excellent starting point for this research. As I continue my work in this area, I am looking forward to working with any other fraud-labeled health insurance claim data that you have or can provide. Any recommendations you might have for other datasets with strong, validated labels would be greatly appreciated.

Thank you again for your valuable contribution to the open-source community!
